asab/api/discovery.py

```diff
@@ -250,6 +259,20 @@ def get_data(item):
 			yield item, json.loads(item_data)
 
 
+	def _on_change(self, watched_event):
+		future = asyncio.run_coroutine_threadsafe(self._rescan_advertised_instances(), self.App.Loop)
```

Comment:
This future should only be run when the previous finished, i. e. self._concurrent_future.done() or self._concurrent_future is None.